RecoTauTag/RecoTau/plugins/DeepTauId.cc

```diff
@@ -1185,59 +1730,64 @@ class DeepTauId : public deep_tau::DeepTauBase {
-    get(dnn::neutralIsoPtSum) = getValueNorm(tau.tauID("neutralIsoPtSum"), 57.59f, 155.3f);
+    get(dnn::chargedIsoPtSum) = getValueNorm(tau_funcs.getChargedIsoPtSum(tau, tau_index, tau_ref), 47.78f, 123.5f);
+    get(dnn::chargedIsoPtSumdR03_over_dR05) = getValue(tau_funcs.getChargedIsoPtSumdR03(tau, tau_index, tau_ref) /
+                                                       tau_funcs.getChargedIsoPtSum(tau, tau_index, tau_ref));
+    get(dnn::footprintCorrection) =
+        getValueNorm(tau_funcs.getFootprintCorrectiondR03(tau, tau_index, tau_ref), 9.029f, 26.42f);
+    get(dnn::neutralIsoPtSum) = getValueNorm(tau_funcs.getNeutralIsoPtSum(tau, tau_index, tau_ref), 57.59f, 155.3f);
     get(dnn::neutralIsoPtSumWeight_over_neutralIsoPtSum) =
-        getValue(tau.tauID("neutralIsoPtSumWeight") / tau.tauID("neutralIsoPtSum"));
+        getValue(tau_funcs.getNeutralIsoPtSumWeight(tau, tau_index, tau_ref) /
+                 tau_funcs.getNeutralIsoPtSum(tau, tau_index, tau_ref));
     get(dnn::neutralIsoPtSumWeightdR03_over_neutralIsoPtSum) =
-        getValue(tau.tauID("neutralIsoPtSumWeightdR03") / tau.tauID("neutralIsoPtSum"));
-    get(dnn::neutralIsoPtSumdR03_over_dR05) = getValue(tau.tauID("neutralIsoPtSumdR03") / tau.tauID("neutralIsoPtSum"));
+        getValue(tau_funcs.getNeutralIsoPtSumdR03Weight(tau, tau_index, tau_ref) /
+                 tau_funcs.getNeutralIsoPtSum(tau, tau_index, tau_ref));
+    get(dnn::neutralIsoPtSumdR03_over_dR05) = getValue(tau_funcs.getNeutralIsoPtSumdR03(tau, tau_index, tau_ref) /
+                                                       tau_funcs.getNeutralIsoPtSum(tau, tau_index, tau_ref));
     get(dnn::photonPtSumOutsideSignalCone) =
-        getValueNorm(tau.tauID("photonPtSumOutsideSignalConedR03"), 1.731f, 6.846f);
-    get(dnn::puCorrPtSum) = getValueNorm(tau.tauID("puCorrPtSum"), 22.38f, 16.34f);
+        getValueNorm(tau_funcs.getPhotonPtSumOutsideSignalCone(tau, tau_index, tau_ref), 1.731f, 6.846f);
+    get(dnn::puCorrPtSum) = getValueNorm(tau_funcs.getPuCorrPtSum(tau, tau_index, tau_ref), 22.38f, 16.34f);
     // The global PCA coordinates were used as inputs during the NN training, but it was decided to disable
     // them for the inference, because modeling of dxy_PCA in MC poorly describes the data, and x and y coordinates
     // in data results outside of the expected 5 std. dev. input validity range. On the other hand,
     // these coordinates are strongly era-dependent. Kept as comment to document what NN expects.
     if (!disable_dxy_pca_) {
-      get(dnn::tau_dxy_pca_x) = getValueNorm(tau.dxy_PCA().x(), -0.0241f, 0.0074f);
-      get(dnn::tau_dxy_pca_y) = getValueNorm(tau.dxy_PCA().y(), 0.0675f, 0.0128f);
-      get(dnn::tau_dxy_pca_z) = getValueNorm(tau.dxy_PCA().z(), 0.7973f, 3.456f);
+      get(dnn::tau_dxy_pca_x) = getValueNorm(tau_funcs.getdxyPCA(tau, tau_index).x(), -0.0241f, 0.0074f);
```

Comment:
```suggestion
      auto const pca = tau_funcs.getdxyPCA(tau, tau_index);
      get(dnn::tau_dxy_pca_x) = getValueNorm(pca.x(), -0.0241f, 0.0074f);
```
to avoid recomputation below